### PR TITLE
Clear test dependency management by create azkaban-db-{api, test} modules

### PR DESCRIPTION
**Pull request by kunkun-tang**
This PR is a follow-up of #1361 , which aimed at clearing the submodule dependency management. Change being made in this pr:

* create azkaban-db-test and azkaban-db-api sub modules.
* move DB's `abstract` class to  azkaban-db-api , and move Test class to `azkaban-db-test`. In this way, we can call up azkaban-db-test in `azkaban-common` module.
* azkaban-db will rely on azkaban-db-api.

### CODE REVIEW

File: azkaban-common/build.gradle [hunk 1]
```diff
@@ -34,6 +34,7 @@ model {
 dependencies {
     compile project(':azkaban-spi')
     compile project(':azkaban-db')
+    compile project(':azkaban-db-test')
```
Comment (HappyRay) [inline azkaban-common/build.gradle]: Shouldn't it be testCompile?

File: azkaban-db/build.gradle [hunk 2]
```diff
@@ -18,6 +18,9 @@ apply plugin: 'distribution'
 
 dependencies {
 
+    compile project(':azkaban-db-api')
+    compile project(':azkaban-db-test')
```
Comment (HappyRay) [inline azkaban-db/build.gradle]: Shouldn't it be testCompile?

File: azkaban-common/build.gradle (hunk 1 above)
Comment (kunkun-tang) [inline azkaban-common/build.gradle]: Yes, you are tight!

File: azkaban-db/build.gradle (hunk 2 above)
Comment (kunkun-tang) [inline azkaban-db/build.gradle]: Yes, you are tight!

**Comment by kunkun-tang** [comment]
We can discuss together with @HappyRay

**Pull request closed without merging**